bin/run_docker.py

```diff
@@ -0,0 +1,236 @@
+#!/usr/bin/env python3
+"""
+This module runs the docker container and outputs relevant log messages into a log file
+which then gets stored on Synapse under its corresponding log folder.
+
+This module recycles functionality seen here:
+https://github.com/Sage-Bionetworks-Challenges/model-to-data-challenge-workflow/blob/82ff3dc0ea8b83d727a7fbecf9550efdc010eadd/run_docker.py
+"""
+
+import os
+import sys
+from typing import Optional, Union
+
+import docker
+import synapseclient
+
+from create_folders import create_folders
+
+
+def get_submission_image(submission_id: str) -> str:
+    """
+    Retrieves Docker Image ID from submission
+
+    Arguments:
+        submission_id: Submission ID to be queried
+
+    Returns:
+        image_id: Docker image identifier in the format: '<image_name>@<sha_code>'
+
+    Raises:
+        ValueError: If submission has no associated Docker image
+    """
+    syn = synapseclient.login(silent=True)
```

Comment:
You should look to pass around 1 instance of `syn` everywhere it is needed. The way you have the script now is that you'll log in twice when you only need to log in once.

1. Login here
2. Login when you call `create_folders`

**Or** at the entry point to your script use login once - The will set a "global" instance of the synapse_client on the `Synapse` class. You'll be able to retrieve your logged in instance with:
```
from synapseclient import Synapse

def func_1():
    syn = Synapse.get_client()

def func_2():
    syn = Synapse.get_client()


def main():
    synapseclient.login(silent=True)
    func_1()
    func_2()
```

Reply:
I can move the login to the Synapse client to `main()`, however I still have to log in again when calling `create_folders`. I don't have multiple functions in `run_docker.py` calling `syn`, it's just `get_submission_image` and then the calls to `create_folders` which is the main function in `create_folders.py` which _does_ login to syn. So the solution proposed wouldn't solve that problem unless I do a major refactoring. Am I understanding your suggestion correctly?

Also I don't like the idea of importing `synapseclient` and then `synapseclient import Synapse` so I can do something like this:

```
import synapseclient

def get_submission_image(syn, **kwargs):

def main(**kwargs):
    syn = synapseclient.login()
    get_submission_image(syn, **kwargs)
```

Does this work or is there a preference for using `Synapse.get_client()`?

Reply:
> Does this work or is there a preference for using Synapse.get_client()?

This suggestion works fine. Doing:
```
import synapseclient

def get_submission_image(syn, **kwargs):

def main(**kwargs):
    syn = synapseclient.login()
    get_submission_image(syn, **kwargs)
```

Is generally the accepted way of passing around the syn instance. 


`Synapse.get_client()` is new - I added this with the OOP refactor so I can get the logged-in syn instance in all of the models without folks needing to manually pass the logged-in instance around.

Reply:
> I don't have multiple functions in run_docker.py calling syn, it's just get_submission_image and then the calls to create_folders which is the main function in create_folders.py which does login to syn.

To be clear on this part:

1. Login once in `run_docker`
2. Pass `syn` to `get_submission_image`
3. Pass `syn` to `store_log_file` which then passes it to `create_folders`
4. Pass `syn` to `store_predictions_file` which then passes it to `create_folders`

Reply:
Also to mention - If `create_folders` is called before you'll have a logged in syn instance you could do:
```
def create_folders(syn, **kwargs):
    if syn is None:
        syn = synapseclient.login()
    .
    .
    .
```

This way calling it from `run_docker.py` is satisfied since it is already logging in. And `create_folders.py` will handle if you didn't already have a syn instance